## File "./MenhirLib/Interpreter_complete.v", line 791, characters 28-38:
## Error: The variable plus_assoc was not found in the current environment.

## File "./MenhirLib/Interpreter_complete.v", line 819, characters 10-20:
## Error: The variable le_lt_n_Sm was not found in the current environment.

Index: MenhirLib/Interpreter_complete.v
--- MenhirLib/Interpreter_complete.v.orig
+++ MenhirLib/Interpreter_complete.v
@@ -788,9 +788,9 @@ Proof.
   - assert (Hptd := next_ptd_cost ptd). destruct next_ptd=>//. by rewrite Hptd.
   - rewrite Nat.add_0_r. assert (IH1 := IH ptd). destruct next_ptd_iter as [ptd'|].
     + specialize (IH ptd'). destruct next_ptd_iter as [ptd''|].
-      * by rewrite IH1 IH -!plus_assoc.
-      * rewrite IH1. by apply plus_lt_compat_l.
-    + by apply lt_plus_trans.
+      * by rewrite IH1 IH -!Nat.add_assoc.
+      * rewrite IH1. by apply Nat.add_lt_mono_l.
+    + by apply Nat.lt_le_trans with (1 := IH1), Nat.le_add_r.
 Qed.
 
 (** We now prove the top-level parsing function. The only thing that
@@ -816,8 +816,8 @@ Proof.
   destruct next_ptd_iter.
   - destruct Hparse as (? & -> & ?). apply (f_equal S) in Hcost.
     rewrite -ptd_cost_build_from_pt Nat.add_0_r in Hcost. rewrite Hcost.
-    apply le_lt_n_Sm, le_plus_l.
-  - rewrite Hparse. split; [|split]=>//. apply lt_le_S in Hcost.
+    apply Nat.lt_succ_r, Nat.le_add_r.
+  - rewrite Hparse. split; [|split]=>//. apply Nat.le_succ_l in Hcost.
     by rewrite -ptd_cost_build_from_pt Nat.add_0_r in Hcost.
 Qed.
 
